src/extension/app/components/onboarding/onboarding-dialog.js

```diff
@@ -0,0 +1,260 @@
+/**
+ * The onboarding item type
+ * @typedef {import('@Types').OnboardingItem} OnboardingItem
+ */
+
+/**
+ * Spectrum tabs
+ * @typedef {import('@spectrum-web-components/tabs').Tabs} Tabs
+ */
+
+/**
+ * Spectrum dialog
+ * @typedef {import('@spectrum-web-components/dialog').DialogBase} DialogBase
+ */
+
+const TOOLS_ORIGIN = 'https://tools.aem.live';
+
+export class OnBoardingDialog extends ConnectedElement {
+  /**
+   * @type {Promise<Tabs>}
+   */
+  @queryAsync('sp-tabs')
+  accessor tabs;
+
+  /**
+   * @type {Promise<DialogBase>}
+   */
+  @queryAsync('sp-dialog-base')
+  accessor dialog;
+
+  /**
+   * The onboarding items
+   * @type {OnboardingItem[]}
+   */
+  @state()
+  accessor items;
+
+  /**
+   * The theme
+   * @type {'dark' | 'light'}
+   */
+  @state()
+  accessor theme;
+
+  /**
+   * The selected index
+   * @type {number}
+   */
+  @state()
+  accessor selectedIndex = 0;
+
+  static get styles() {
+    return [style];
+  }
+
+  async connectedCallback() {
+    super.connectedCallback();
+
+    this.fetchIndex();
+    // Listen for theme changes
+    window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', this.onThemeChange);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    window.matchMedia('(prefers-color-scheme: dark)').removeEventListener('change', this.onThemeChange);
+  }
+
+  /**
+   * Handle theme changes
+   */
+  onThemeChange = () => {
+    /* istanbul ignore next */
+    this.requestUpdate();
+  };
+
+  /**
+   * Fetch the onboarding index
+   */
+  async fetchIndex() {
+    const resp = await fetch(`${TOOLS_ORIGIN}/sidekick/query-index.json`);
+    if (!resp.ok) {
+      this.remove();
+      return;
+    }
+
+    const index = await resp.json();
+    const onboardingItems = index.data
+      .filter(({ category }) => category === 'onboarding')
+      .map(async (item) => {
+        const fetchResp = await fetch(`${TOOLS_ORIGIN}${item.path}.plain.html`);
+        return fetchResp.ok
+          ? { ...item, content: await fetchResp.text() }
+          : null;
+      });
+
+    this.items = (await Promise.all(onboardingItems)).filter(Boolean);
+  }
+
+  /**
+   * Handle tab changes
+   * @param {*} event
+   */
+  async onTabChange(event) {
+    this.selectedIndex = Number(event.target.value);
+  }
+
+  /**
+   * Handle next button clicks
+   */
+  async onNextClicked() {
+    // If the user is on the last item, close the dialog
+    if (this.selectedIndex + 1 >= this.items.length) {
+      this.closeDialog();
+      return;
+    }
+
+    // Otherwise, move to the next item
+    this.selectedIndex += 1;
+  }
+
+  /**
+   * Handle action button clicks
+   */
+  async onActionClicked() {
+    const { action } = this.items[this.selectedIndex];
+
+    const actionsMap = {
+      import: () => {
+        this.appStore.sampleRUM('click', { source: 'sidekick', target: 'onboard-modal:import-projects' });
+        chrome.runtime.sendMessage({ action: 'importProjects' });
```

Comment:
@rofe Would be great to avoid the dialogs on-top of dialogs. IMO we should handle this all using chrome notifications.

If no legacy sidekick found, show notification "No legacy sidekick found, if the extension is installed but disabled, enable it and try again."

wdyt?

> they can import their projects at a later time via the context menu
I think this should be something for the onboarding blurb for this item.

Reply:
Yeah, depending on the rules set on your focus mode on macOS they may or may not be shown.

Reply:
I think we should drop native notifications then completely.

Reply:
We don't currently have an alternative for actions happening in the background script (i.e. context menu).

Reply:
Should now use native "sidekick" notifications (dialogs). More details on how it works in the PR description.